src/Appwrite/Services/Functions.cs

```diff
@@ -510,15 +562,18 @@ public Task<Models.ExecutionList> ListExecutions(string functionId, List<string>
         /// function execution process will start asynchronously.
         /// </para>
         /// </summary>
-        public Task<Models.Execution> CreateExecution(string functionId, string? data = null, bool? xasync = null)
+        public Task<Models.Execution> CreateExecution(string functionId, string? body = null, bool? xasync = null, string? xpath = null, string? method = null, object? headers = null)
         {
-            var path = "/functions/{functionId}/executions"
+            var apiPath = "/functions/{functionId}/executions"
                 .Replace("{functionId}", functionId);
 
             var parameters = new Dictionary<string, object?>()
             {
-                { "data", data },
-                { "async", xasync }
+                { "body", body },
+                { "async", xasync },
+                { "path", xpath },
+                { "method", method },
+                { "headers", headers }
             };
 
             var headers = new Dictionary<string, string>()
```

Comment:
If `headers` is being added as a function parameter, initializing it will throw an error.

Also, if we need to add a null check, we need to add that before `var parameters` is initialized.